pkg/cmd/pipeline/start.go

```diff
@@ -765,6 +765,15 @@ func parsePipeline(taskLocation string, p cli.Params) (*v1beta1.Pipeline, error)
 	if err := yaml.UnmarshalStrict(b, &pipeline); err != nil {
 		return nil, err
 	}
+
+	for _, param := range pipeline.Spec.Params {
+		if param.Type == "" {
+			err := fmt.Errorf("The param '%s' does not have a type", param.Name)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
```

Comment:
Wondering if all errors for not having a `type` specified should be displayed at once instead of returning the error on first failure. So if two params are specified without types, tkn should show both errors for both params.

Reply:
@danielhelfand , Yes, and what I am also thinking is that currently I am comparing this with an empty string, but it may be possible that the type is invalid(other than string/array), I think in that case also we should give an error for each param of course.